Src/Sentry.Xamarin.Forms/Internals/FormsNavigationIntegration.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Sentry.Xamarin.Forms.Extensions;
+using Sentry.Xamarin.Internals;
+using System.Collections.Generic;
+using Xamarin.Forms;
+
+namespace Sentry.Xamarin.Forms.Internals
+{
+    internal class FormsNavigationIntegration : IPageNavigationTracker
+    {
+        public string CurrentPage { get; private set; }
+        private IHub _hub { get; set; }
+
+        private bool _disabled { get; set; }
+
+        /// <summary>
+        /// Registers the Page appearing and disappearing event to the given application..
+        /// </summary>
+        /// <param name="application">The Xamarin Application.</param>
+        internal void ApplySentryNavigationEvents(Application application)
+        {
+            if (!_disabled)
+            {
+                application.PageAppearing += Current_PageAppearing;
+                application.PageDisappearing += Current_PageDisappearing;
+            }
+        }
+
+        public void Register(IHub hub, SentryOptions _) => _hub = hub;
+
+        /// <summary>
+        /// Removes the PageDissapearing and PageAppearing events from the main Application, if found.
+        /// </summary>
+        public void Unregister()
+        {
+            if (!_disabled)
+            {
+                _disabled = true;
```

Comment:
Are there potentially multiple simultaneous threads calling `Unregister` and `ApplySentryNavigationEvents`?  (I'm not sure.)

If so, consider marking `_disabled` with `volatile` or using `Interlocked` operations, etc.